File: VB/MVVMExpenses/Models/MyDbContextDataModel/IMyDbContextUnitOfWork.vb
Imports System
Imports System.Linq
Imports System.Data
Imports System.Data.Entity
Imports System.Linq.Expressions
Imports System.Collections.Generic
Imports Common.Utils
Imports Common.DataModel
Imports Common.DataModel.EntityFramework
Imports MVVMExpenses.DataBase
Imports MVVMExpenses.DataModels
Namespace MVVMExpenses.Models.MyDbContextDataModel
    ''' <summary>
    ''' IMyDbContextUnitOfWork extends the IUnitOfWork interface with repositories representing specific entities.
    ''' </summary>
    Public Interface IMyDbContextUnitOfWork
        Inherits IUnitOfWork
        ''' <summary>
        ''' The Account entities repository.
        ''' </summary>
        ReadOnly Property Accounts As IRepository(Of Account, Long)
        ''' <summary>
        ''' The Category entities repository.
        ''' </summary>
        ReadOnly Property Categories As IRepository(Of Category, Long)
        ''' <summary>
        ''' The Transaction entities repository.
        ''' </summary>
        ReadOnly Property Transactions As IRepository(Of Transaction, Long)
    End Interface
End Namespace

File: CS/MVVMExpenses/Models/MyDbContextDataModel/IMyDbContextUnitOfWork.cs
using System;
using System.Linq;
using System.Data;
using System.Data.Entity;
using System.Linq.Expressions;
using System.Collections.Generic;
using MVVMExpenses.Common.Utils;
using MVVMExpenses.Common.DataModel;
using MVVMExpenses.Common.DataModel.EntityFramework;
using MVVMExpenses.DataBase;
using MVVMExpenses.DataModels;

namespace MVVMExpenses.Models.MyDbContextDataModel {
    /// <summary>
    /// IMyDbContextUnitOfWork extends the IUnitOfWork interface with repositories representing specific entities.
    /// </summary>
    public interface IMyDbContextUnitOfWork : IUnitOfWork {

        /// <summary>
        /// The Account entities repository.
        /// </summary>
        IRepository<Account, long> Accounts { get; }

        /// <summary>
        /// The Category entities repository.
        /// </summary>
        IRepository<Category, long> Categories { get; }

        /// <summary>
        /// The Transaction entities repository.
        /// </summary>
        IRepository<Transaction, long> Transactions { get; }
    }
}
